game.py

```diff
@@ -687,35 +693,30 @@ def _footer(self):
         return '</svg>\n'
 
     def save_highscore(self):
-        file_path = os.path.join(self._activity_root, 'data', 'highscore')
+        file_path = os.path.join(get_activity_root(), 'data', 'highscore')
         logging.debug(file_path)
-        highscore = []
-        highscore.append(0)
+        highscore = [0]
         if os.path.exists(file_path):
-            File = open(file_path, "r")
-            highscore = File.readlines()
-            File.close()
+            with open(file_path, "r") as fp:
+                highscore = fp.readlines()
 
         int_highscore = int(highscore[0])
-        if not (int_highscore > self._correct):
-            File = open(file_path, "w")
-            File.write(str(self._correct))
-            File.close()
+        if not int_highscore > self._correct:
+            with open(file_path, "w") as fp:
+                fp.write(str(self._correct))
 
 
     def load_highscore(self):
-        file_path = os.path.join(self._activity_root, 'data', 'highscore')
+        file_path = os.path.join(get_activity_root(), 'data', 'highscore')
         logging.debug(file_path)
         if os.path.exists(file_path):
             try:
-                File = open(file_path, "r")
-                highscore = int(File.readlines()[0])
-                File.close()
-                return highscore
-            except BaseException:
+                with open(file_path, "r") as fp:
+                    highscore = fp.readlines()
+                return int(highscore[0])
+            except ValueError and IndexError:
```

Comment:
Shouldn't the `and` here be an `or` as both errors are different?

Reply:
yes, I'll get it right